### DISCUSSION TITLE
Is it possible to have 1 to 1 relations with multiple fields where either field matches?

### DISCUSSION THREAD

**Discussion by cryptrr** [prisma/prisma#23823] [Q&A]
### Question

I would like to know if this scenario is possible with Prisma.

I have two models Product and BaseProduct.
Is it possible to have a 1 to 1 connection from Product to BaseProduct when either upc or ean matches the BaseProducts upc or ean.

```
model Product {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now()) @unique

  upc String? @unique
  ean String? @unique
  baseProduct BaseProduct? @relation(fields: [upc], references: [upc])
}
```

```
model BaseProduct {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())

  upc String? @unique
  ean String?  @unique
}
```

**Comment by ludralph** [comment]
Hi @cryptrr 👋 

Thank you for raising this question.

It's not directly possible to have a one-to-one relation in Prisma Schema based on either of two fields (`upc` or `ean`). In Prisma, one-to-one relation is established based on a single unique field.

You're trying to establish a relation from `Product` to `BaseProduct` where either `upc` or `ean` matches. Prisma Schema currently does not support this kind of multi-field or conditional referencing in relations.

What you can do however is to create two separate one-to-one relations from `Product` to `BaseProduct` - one based on `upc` and another based on `ean`. For example:

```prisma
model Product {
  id             String       @id @default(cuid())
  createdAt      DateTime     @unique @default(now())
  upc            String?      @unique
  ean            String?      @unique
  baseProductUPC BaseProduct? @relation("BaseProductUPC", fields: [upc], references: [upc])
  baseProductEAN BaseProduct? @relation("BaseProductEAN", fields: [ean], references: [ean])
}

model BaseProduct {
  id         String   @id @default(cuid())
  createdAt  DateTime @default(now())
  upc        String?  @unique
  ean        String?  @unique
  productUPC Product? @relation("BaseProductUPC")
  productEAN Product? @relation("BaseProductEAN")
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
